docs/sources/operations/parquet/schema.md

```diff
@@ -0,0 +1,274 @@
+          required binary Name (STRING);
+          repeated group Attrs {
+            required binary Key (STRING);
+            required binary Value;
+          }
+          required int32 DroppedAttributesCount (INTEGER(32,true));
+          optional binary Test (STRING);
+        }
+        required int32 DroppedEventsCount (INTEGER(32,true));
+        required binary Links;
+        required int32 DroppedLinksCount (INTEGER(32,true));
+
+        optional binary HttpMethod (STRING);
+        optional binary HttpUrl (STRING);
+        optional int64 HttpStatusCode (INTEGER(64,true));
+      }
+    }
+  }
+}
+```
+## Trace-level attributes
+For speed and ease-of-use, we are projecting several values to columns at the trace-level:
+* Trace ID - Don't store on each span.
+* Root service/span names/StartTimeUnixNano - These are selected properties of the root span in each trace (if there is one). These are used for displaying results in the Grafana UI. These properties are computed at ingest time and stored once for efficiency, so we don't have to find the root span.
+* `DurationNanos` - The total trace duration, computed at ingest time. This powers the min/max duration filtering in the current Tempo search and is more efficient than scanning the spans duration column. However, it may go away with TraceQL or we could decide to change it to span-level duration filtering too.
+
+## Dedicated columns
+
+Projecting attributes to their own columns has benefits for speed and size.
+Therefore we are taking an opinionated approach and projecting some common attributes to their own columns.
+All other attributes are stored in the generic key/value maps and are still searchable, but not as quickly.
+We chose these attributes based on what we commonly use ourselves (scratching our own itch), but we think they will be useful to most workloads.
+
+Resource-level attributes include the following:
+
+* `service.name`
+* `cluster` and `k8s.cluster.name`
+* `namespace` and `k8s.namespace.name`
+* `pod` and `k8s.pod.name`
+* `container` and `k8s.container.name`
+
+Span-level attributes include the following:
+
+* `http.method`
+* `http.url`
+* `http.status_code` (int)
+
+## "Any"-type Attributes
+
+OTLP attributes have variable data types, which is easy to accomplish in formats like protocol-buffers, but does not translate directly to Parquet.
+Each column must have a concrete type.
+There are several possibilities here but we chose to have optional values for each concrete type.
+`Array` and `KeyValueList` types are stored as protocol-buffer-encoded byte arrays.
+
+```yaml
+repeated group Attrs {
+    required binary  Key (STRING);
+
+    # Only one of these will be set
+    optional binary  Value (STRING);
+    optional boolean ValueBool;
+    optional double  ValueDouble;
+    optional int64   ValueInt (INT(64,true));
+    optional binary  ValueArray (STRING);
+    optional binary  ValueKVList (STRING);
+}
+```
+
+Pros:
```

Comment:
Can probably remove historical discussion of pros/cons

Reply:
Thank you. I've removed it.